--- qtdeclarative-everywhere-src-5.11.1.orig/src/quick/items/qquickloader.cpp
+++ qtdeclarative-everywhere-src-5.11.1/src/quick/items/qquickloader.cpp
@@ -629,6 +629,25 @@
     }
 }
 
+QString QQuickLoader::errorString() const
+{
+   Q_D(const QQuickLoader);
+   QString ret;
+   if(d->component->errors().isEmpty())
+      return ret;
+
+   if (d->component && d->component->isError()) {
+        const QList<QQmlError> errorList = d->component->errors();
+        for (const QQmlError &e : errorList) {
+            ret += e.url().toString() + QLatin1Char(':') +
+              QString::number(e.line()) + QLatin1Char(' ') +
+              e.description() + QLatin1Char('\n');
+        }
+   }
+ 
+   return ret;
+}
+
 void QQuickLoaderIncubator::setInitialState(QObject *o)
 {
     loader->setInitialState(o);
--- qtdeclarative-everywhere-src-5.11.1.orig/src/quick/items/qquickloader_p.h
+++ qtdeclarative-everywhere-src-5.11.1/src/quick/items/qquickloader_p.h
@@ -76,6 +76,7 @@
     void setActive(bool newVal);
 
     Q_INVOKABLE void setSource(QQmlV4Function *);
+    Q_INVOKABLE QString errorString() const;
 
     QUrl source() const;
     void setSource(const QUrl &);
